### ISSUE TITLE
Getting Node low level errors running vitest

### ISSUE THREAD

**Issue by lemoustachiste** [vitest-dev/vitest#3816]
### Describe the bug

So I am testing migration of my project from jest to vitest.

But when I run `vitest run` through `npm run test`, I keep getting Node errors such as follows:

```
node(39994,0x7000083d9000) malloc: *** error for object 0x600002c44cd0: pointer being freed was not allocated
node(39994,0x700006fca000) malloc: *** error for object 0x600002c44cd0: pointer being freed was not allocated
node(39994,0x7000083d9000) malloc: *** set a breakpoint in malloc_error_break to debug
node(39994,0x700006fca000) malloc: *** set a breakpoint in malloc_error_break to debug
```

This error does not even get to a point of running tests. 

Or if some tests start running, then they stop unexpectedly with the following error:

```
[1]    31930 segmentation fault  npm run test
```

Errors occur with both node 16.20.0 and 18.16.1.

My tests are all TS, so is my project.

Any clue what could be causing this?

### Reproduction

Not sure if I can provide a repro at this point.

### System Info

```shell
System:
    OS: macOS 13.5
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 21.56 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 8.12.1 - ~/work/cert-verifier-js/node_modules/.bin/npm
  Browsers:
    Chrome: 115.0.5790.102
    Safari: 16.6
  npmPackages:
    vitest: ^0.33.0 => 0.33.0
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lemoustachiste** [comment]
@sheremet-va would sharing the repo with a work branch for vitest work as a repro for you? On simpler code I cannot replicate the issue.

**Comment by sheremet-va** [comment]
> @sheremet-va would sharing the repo with a work branch for vitest work as a repro for you? On simpler code I cannot replicate the issue.

Yes

**Comment by lemoustachiste** [comment]
Please have a look at this branch: https://github.com/blockchain-certificates/cert-verifier-js/tree/chore/vitest

just:

- `npm i`
- `npm test` (single run)

I did not get the mem alloc failure just now, but it still does not complete with `segmentation fault` at unexpected stages it looks like.

I haven't investigated much yet, this is exploratory work to try and move away from jest. 

Thanks

**Comment by sheremet-va** [comment]
This error type usually happens when an application cannot work in multi-threaded mode. In this case, it is recommended to disable `threads`: run `vitest --no-threads` instead of `vitest`.

**Comment by AriPerkkio** [comment]
Fixed by https://github.com/vitest-dev/vitest/pull/3925 and https://github.com/vitest-dev/vitest/pull/4172.

Vitest 1.0.0-beta has now `--pool=forks` which uses multiple `node:child_process` in parallel. The previous `--no-threads` was using a single `node:child_process`.

If the code you are testing is incompatible with `node:worker_threads`, switch to `--pool=forks`. If you run into same error with that pool, feel free to open new issue with minimal reproduction.
